src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String letter;
+
+        Operation(String operation) {
+            this.letter = operation;
+        }
+
+        public String getLetter() {
+            return letter;
+        }
+
+        public static FruitTransaction.Operation getOperation(String letter) {
+            return Arrays.stream(FruitTransaction.Operation.values())
+                    .filter(a -> a.getLetter().equals(letter))
+                    .findFirst()
+                    .orElseThrow(
+                            () -> new IllegalArgumentException(
+                                    "No enum constant for letter: " + letter
+                            )
+                    );
+        }
```

Comment:
The `getOperation` method throws an `IllegalArgumentException` if no matching operation is found. This is a good practice for handling invalid inputs. Ensure that this exception is handled appropriately wherever this method is used to prevent unexpected crashes.